lib/reqd.ml

```diff
@@ -51,7 +51,7 @@ type response_state =
       ; mutable iovec :
           [ `String of string | `Bigstring of Bigstringaf.t ] Httpaf.IOVec.t
       }
-  | Streaming of Response.t * [ `write ] Body.t
+  | Streaming of Response.t * [ `write ] Body.t * Headers.t option
```

Comment:
I think a triple is probably the cutoff in my mind for preferring a record.

I'd rather see a record (can probably be inline) here, and I think we don't necessarily need the type of trailers to be optional. Initializing it with `Headers.empty` and comparing to that will probably save allocations.